dvc/repo/imp.py

```diff
@@ -8,12 +8,11 @@ def imp(self, url, out, resume=False):
 
     self.check_dag(self.stages() + [stage])
 
-    self.files_to_git_add = []
```

Comment:
But you are not using the decorator for imp().

Reply:
That is true, however take a look at stage.dump(), there is code for file tracking. Even without scm_context, files_to_git_add is unnecessary here.

Reply:
As discussed, wrapping with scm_context.